cli/ssh_test.go

```diff
@@ -31,8 +31,13 @@ import (
 	"github.com/coder/coder/testutil"
 )
 
-func setupWorkspaceForAgent(t *testing.T) (*codersdk.Client, codersdk.Workspace, string) {
+func setupWorkspaceForAgent(t *testing.T, mutate func([]*proto.Agent) []*proto.Agent) (*codersdk.Client, codersdk.Workspace, string) {
```

Comment:
Maybe `mutate ...func`? You would get rid of the `nil` in `setupWorkspaceForAgent(t, nil)`.

Reply:
Clean suggestion, but I think we can keep this explicit for now, makes it slightly more convenient to add new arguments, if needed and I don't see a use-case for multiple mutation functions.

Reply:
In general, the nil is considered to be an anti-pattern as it has an ambiguous meaning. This is the main reason why I raised this comment, so feel free to leave it as is.